crates/sqlbuiltins/src/functions/scalars/hashing.rs

```diff
@@ -0,0 +1,100 @@
+use std::hash::{Hash, Hasher};
+
+use fnv::FnvHasher;
+use siphasher::sip::SipHasher24;
+
+use super::*;
+
+pub struct SipHash;
+
+impl ConstBuiltinFunction for SipHash {
+    const NAME: &'static str = "siphash";
```

Comment:
Some of our other functions such as `digest` have them all wrapped up into a single function with an algorithm argument. Wondering if we could/should do the same here. `hash(<value>, 'siphash' | 'fnv')`. I'm not partial to either way, just a thought.

Reply:
In the digest case the `digest` functionality is also provided by dedicated methods (like the ones in this PR.) 

My assumption is that the `digest` API mostly exists so you could (say) store or compute which hash algorithm to use, and therefore select the algorithm at runtime. This is a pretty narrow use case, and maybe makes sense for cryptographic hashes (maybe?) but makes much less sense for non-cryptographic hashes, but it would be easy to add later if someone wanted it.